docs/adding-new-resource.md

```diff
@@ -0,0 +1,705 @@
+
+4. Check if the resource is an SDK resource or Framework resource from the
+[source code], for SDK resources, you will see a comment line like 
+`// @SDKResource` in the source code:
+
+- The `aws_redshift_endpoint_access` resource is an SDK resource, go to
+`config/externalname.go` and add the following line to the
+`TerraformPluginSDKExternalNameConfigs` table:
+
+  - Check the `redshift` group, if there is a group, add the external-name config below:
+  ```golang
+  // redshift
+  ...
+  // Redshift endpoint access can be imported using the endpoint_name
+  "aws_redshift_endpoint_access": config.ParameterAsIdentifier("endpoint_name"),
+  ```
+  - If there is no group, continue by adding the group name as a comment line.
+
+- For Framework resources, you will see a comment line like 
+`// @FrameworkResource` in the source code. If the resource is a Framework
+resource, add the external-name config to the
+`TerraformPluginFrameworkExternalNameConfigs` table.
+
+*Note: Look at the `config/externalnamenottested.go` file and check if there is
+a configuration for the resource and remove it from there.*
+
+5. Run `make submodules` to initialize the build submodule and run 
+`make generate`. When the command process is completed, you will see that the
+controller, CRD, generated example, and other necessary files for the resource
+have been created and modified.
+
+```bash
+> git status
+On branch add-redshift-endpoint-access
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git restore <file>..." to discard changes in working directory)
+	modified:   apis/redshift/v1beta1/zz_generated.conversion_hubs.go
+	modified:   apis/redshift/v1beta1/zz_generated.deepcopy.go
+	modified:   apis/redshift/v1beta1/zz_generated.managed.go
+	modified:   apis/redshift/v1beta1/zz_generated.managedlist.go
+	modified:   apis/redshift/v1beta1/zz_generated.resolvers.go
+	modified:   config/externalname.go
+	modified:   config/externalnamenottested.go
+	modified:   config/generated.lst
+	modified:   internal/controller/zz_monolith_setup.go
+	modified:   internal/controller/zz_redshift_setup.go
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+	apis/redshift/v1beta1/zz_endpointaccess_terraformed.go
+	apis/redshift/v1beta1/zz_endpointaccess_types.go
+	examples-generated/redshift/v1beta1/endpointaccess.yaml
+	internal/controller/redshift/endpointaccess/
+	package/crds/redshift.aws.upbound.io_endpointaccesses.yaml
+```
+
+6. Go through the "Warning" boxes (if any) in the Terraform Registry page to
+see whether any of the fields are represented as separate resources as well.
+It usually goes like this:
+
+    ```
+    Routes can be defined either directly on the azurerm_iothub
+    resource, or using the azurerm_iothub_route resource - but the two cannot be
+    used together.
+    ```
+    In such cases, the field should be moved to status since we prefer to
```

Comment:
Yes, but since the only place where the field can be moved in status is `atProvider` and we use `config.MoveToStatus` to move it, I think it can be kept this way to avoid confusion.